### PR TITLE
[Web] Add descriptions to session join modes menu

### LINKED ISSUE #21208: [Web] Add tooltip to explain the different Moderated Session join modes
After support for Moderated Sessions was added in the UI, users now see up to 3 join mode options they can choose from when joining an active session (`Observer`, `Moderator`, `Peer`). However, there is no information in the UI that actually gives context or explains what the join modes are and the differences between them. 

There should be a tooltip somewhere that explains the different join modes and what they can do.

### PR DESCRIPTION
**Pull request by rudream**
## Purpose 

This PR closes https://github.com/gravitational/teleport/issues/21208
Note: The linked issue mentions a tooltip, however we decided that instead of having a tooltip, we would have a short description under each mode that explains it instead.

Adds a short description under each participant mode that briefly describes its abilities. Also, participant modes that you don't have access to are no longer hidden but are now instead greyed out and have a disclaimer.

### Notes

This design slightly deviates from the [figma mockup](https://www.figma.com/proto/8DQZl1mYUPLgvKgmA6zPOM/In-Progress-Work?page-id=4254%3A71594&node-id=4292-74417&viewport=217%2C942%2C0.48&scaling=min-zoom) in order to keep it more consistent with the rest of our design system (see our popover menus for connecting to Nodes, Desktops or AWS Console Apps).

## Demo

image image

image image

image image

### CODE REVIEW

**Review by zmb3 — COMMENTED** [review]
Code looks fine, but the disabled state could use a little more contrast IMO.

I'm not an expert here though. If @roraback is good with it, then 🚢

File: web/packages/teleport/src/Sessions/SessionList/SessionJoinBtn.tsx [hunk 1]
```diff
@@ -82,28 +81,91 @@ function JoinMenu({ children }: { children: React.ReactNode }) {
...
+      `}
+    >
+      <Box height="fit-content" width="264px">
+        <Text typography="h6">{title}</Text>
+        <Text>{description}</Text>
+        <Box color="text.primary" px={1} mt={1}>
+          <Text fontSize="10px" color="text.secondary">
+            <Warning color="error.main" mr={2} />
+            You do not have permission to join as a
+            {participantMode === 'observer' ? 'n ' : ' '}
```
Comment (avatus) [inline web/packages/teleport/src/Sessions/SessionList/SessionJoinBtn.tsx:158]: This is clever. Another way to maybe do this is have an "warning map" or something like
```js
const WARNING_TEXT = {
    observer: "You do not have permission to join as an observer",
    moderator: "You do not have permission to join as a moderator",
    peer: "You do not have permission to join as a peer"
}

then

{WARNING_TEXT[participantMode]}

```

not necessary but just offering another option. You pick.

**Review by avatus — APPROVED** [review]

**Review by ryanclark — APPROVED** [review]

**Comment by rudream** [comment]
> but the disabled state could use a little more contrast IMO.

@zmb3 After https://github.com/gravitational/teleport/pull/25119 was merged, the opacity we use for disabled text increased (from 18% to 36%) which now gives that text more contrast.

image image

**Comment by roraback** [comment]
> Code looks fine, but the disabled state could use a little more contrast IMO.
> 
> I'm not an expert here though. If @roraback is good with it, then 🚢

Technically it's a bit below contrast ratios recommended by WCAG AA standard, but text for disabled options often is. I'm good with it as is.

**Pull request merged**